ni/abc.py

```diff
@@ -101,10 +101,9 @@ class CLAHost(abc.ABC):
 
     @abc.abstractmethod
     async def check(self, client: aiohttp.ClientSession,
-                    usernames: AbstractSet[str]) -> Status:
-        """Check if all of the specified usernames have signed the CLA."""
-        # While it would technically share more specific information if a
-        # mapping of {username: Status} was returned, the vast majority of
-        # cases will be for a single user and thus not worth the added
-        # complexity to need to worry about it.
-        return Status.username_not_found  # pragma: no cover
+                    usernames: AbstractSet[str]) -> Mapping[Status, AbstractSet[str]]:
+        """Check if all of the specified usernames have signed the CLA.
+
+        Return a Mapping of problems and the associated list of usernames.
+        """
+        return {Status.username_not_found: {'username'}}  # pragma: no cover
```

Comment:
This doesn't make sense anymore as a default value, so it would probably be better to raise `NotImplementedError`.